MIDAS/src/data_logging.cpp

```diff
@@ -20,37 +21,89 @@ ASSOCIATE(Orientation, ID_ORIENTATION)
 
 
 template<typename T>
-void log_reading(LogSink& sink, Reading<T>& reading) {
+void log_reading(Logger& sink, Reading<T>& reading) {
     ReadingDiscriminant discriminant = get_discriminant<T>();
     sink.write((uint8_t*) &discriminant, sizeof(ReadingDiscriminant));
```

Comment:
i did not write this, ask Magilan

Reply:
@redindelible hola señor

Reply:
I personally don't like `reinterpret_cast` because it's so long, and a C-style cast is pretty readable as-is. 

We could do a code style pass of MIDAS after we get everything merged in, to keep everything consistent and change everything to have better style.